core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala

```diff
@@ -1232,24 +1243,16 @@ private[ui] class TaskPagedTable(
 
   override def pageLink(page: Int): String = {
```

Comment:
One problem: the URL that this returns is already encoded, but it ends up being re-encoded when Scala XML substitutes it, so the ampersands wind up as `&amp` in the generated HTML. I'll fix this.